api/src/main/java/pl/kolendateam/dadcard/characterCard/CharacterController.java

```diff
@@ -1,34 +1,143 @@
+    public CharacterDTO create(@RequestBody CharacterDTO characterDTO){
         Character character = new Character(characterDTO.characterName,characterDTO.playerName);
 
+        SavingThrow savingThrow = new SavingThrow(0, 0, 0);
+
+        character.setSavingThrow(savingThrow);
+
         this.characterRepository.save(character);
 
-        return character;
+        return characterDTO;
     }
 
-}
+    @PostMapping(value="{id}/class",consumes = {"application/json"})
+    public CharacterDTO setCharacterClass(@PathVariable int id, @RequestBody ClassPgDTO classPgDTO){
+
+        List<Skills> skillsList = this.skillsRepository.findAll();
+
+        Optional<Character> characterOpt = this.characterRepository.findById(id);
+
+        if(!characterOpt.isPresent()){
+            throw new ResponseStatusException(
+                HttpStatus.NOT_FOUND, "Character Not Found");
+        }
+
+        Character character = characterOpt.get();
+
+        Optional <ClassCharacter> classOpt = this.classRepository.findById(classPgDTO.id);
+
+        if(!classOpt.isPresent()){
+            throw new ResponseStatusException(
+                HttpStatus.NOT_FOUND, "Class Not Found");
+        } 
+
+        ClassCharacter classCharacter = classOpt.get();
+
+        ArrayList<ClassPg> classPgList = character.getClassPgArray();
+
+        character.calculateSkillPointsFirstLevel(classCharacter.getSkillPoints());
+
+        character.createSkillsArray(skillsList);
+
+        ClassPg classPg = new ClassPg(classCharacter.getId(),classCharacter.getName(),1,classCharacter.getSavingThrow());
+         
+        int indexClassInDB = classPg.findIndexInArrayById(classPgList);
+
+        if(indexClassInDB == -1){
+            character.addClassToPgArray(classPg);
+            character.setSkillsTruePgArray(classCharacter.getAvailableSkills());           
+        }else{
+            character.incrementLevelClassForIndex(indexClassInDB);
+        }
+
+        int levelClass = classPg.getLevel();
+
+        if(levelClass == 1){
+            character.addSTLevelOne(classPg);
+        }else{
+            character.incementST();
+        }
+    
+        character.incrementLep();
+
+        character.calculateSkillPoints(classCharacter.getSkillPoints());
+        
+        this.characterRepository.save(character);
+        return new CharacterDTO (character);
+    }
+
+    @PostMapping(value="{id}/skill",consumes = {"application/json"})
+    public CharacterDTO buySkillClass(@PathVariable int id, @RequestBody SkillsDTO skillsDTO){
+
+        Optional<Character> characterOpt = this.characterRepository.findById(id);
+
+        if(!characterOpt.isPresent()){
+            throw new ResponseStatusException(
+                HttpStatus.NOT_FOUND, "Character Not Found");
+        }
+
+        Character character = characterOpt.get();
+
+        int lep = character.getLep();
+
+        character.buySkills(skillsDTO.nameSkill,lep,skillsDTO.skillRank);
```

Comment:
character ma tego lep to po co go tu podawac?

Reply:
lepiej było by się posługiwać id skila

Reply:
i pobierac skila z bazy po id zeby nie dało się dodac skila dupa

Reply:
> character ma tego lep to po co go tu podawac?

racja, poprawiłem

Reply:
> lepiej było by się posługiwać id skila

racja, poprawiłem